e2e/nomostest/portforwarder/port_forwarder.go

```diff
@@ -118,17 +118,19 @@ func (pf *PortForwarder) LocalPort() (int, error) {
 }
 
 func (pf *PortForwarder) setPort(cmd *exec.Cmd, port int, pod string, async bool) {
+	pf.logger.Infof("updating port-forward %s:%d -> %s:%d", pf.pod, pf.localPort, pod, port)
 	if async {
 		pf.mux.Lock()
-		defer pf.mux.Unlock()
-	}
-	pf.logger.Infof("updating port-forward %s:%d -> %s:%d", pf.pod, pf.localPort, pod, port)
-	if pf.setPortCallback != nil {
-		pf.setPortCallback(port, pod)
 	}
 	pf.cmd = cmd
 	pf.localPort = port
 	pf.pod = pod
+	if async {
+		pf.mux.Unlock()
+	}
+	if pf.setPortCallback != nil {
```

Comment:
I dont think we need it now, but one way to make this more flexible, slightly cleaner, and reduce the number of branches is to make the callbacks an array and for loop over it. The only problem with this approach for hooks/callbacks in general is removal. If you need removal and ordering, then an OrderedMap is required.

Reply:
Right now we're just registering a single function, but I see your point to flexility if we need to support multiple callbacks